sdk/core/azure-core/src/main/java/com/azure/core/credential/AzureNamedKey.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.credential;
+
+/**
+ * Represents a credential bag containing the key and the name of the key.
+ *
+ * @see AzureNamedKeyCredential
+ */
+public class AzureNamedKey {
+    private final String name;
+    private final String key;
+
+    AzureNamedKey(String name, String key) {
```

Comment:
didn't make this public ,as its purpose is to extract the credential details currently from AzureNamedKeyCredential.